doc/guide/transactions-and-threading.rst

```diff
@@ -0,0 +1,394 @@
+============================
+Transactions and concurrency
+============================
+
+.. contents::
+
+`Transactions <https://en.wikipedia.org/wiki/Database_transaction>`_
+are a core feature of ZODB.  Much has been written about transactions,
+and we won't go into much detail here.  Transactions provide 2 core
+benefits:
+
+Atomicity
+  When a transaction executes, it succeeds or fails completely. If
+  some data are updated and then an error occurs, causing the
+  transaction to fail, the updates are rolled back automatically. The
+  application using the transactional system doesn't have to undo
+  partial changes.  This takes a significant burden from developers
+  and increases the reliability of applications.
+
+Concurrency
+  Transactions provide a way of managing concurrent updates to data.
+  Different programs operate on the data independently, without having
+  to use low-level techniques to moderate their access. Coordination
+  and synchronization happens via transactions.
+
+
+.. _using-transactions-label:
+
+Using transactions
+==================
+
+All activity in ZODB happens in the context of database connections
+and transactions.  Here's a simple example::
+
+  import ZODB, transaction
+  db = ZODB.DB(None) # Use a mapping storage
+  conn = db.open()
+
+  conn.root.x = 1
+  transaction.commit()
+
+.. -> src
+
+   >>> exec(src)
+
+In the example above, we used ``transaction.commit()`` to commit a
+transaction, making the change to ``conn.root`` permanent.  This is
+the most common way to use ZODB, at least historically.
+
+If we decide we don't want to commit a transaction, we can use
+``abort``::
+
+  conn.root.x = 2
+  transaction.abort()
+
+.. -> src
+
+   >>> exec(src)
+   >>> conn.root.x
+   1
+   >>> conn.close()
+
+In this example, because we aborted the transaction, the value of
+``conn.root.x`` was rolled back to 1.
```

Comment:
That's why I spell it out on words.  I'll add a comment on the abort line as well.